web-registry/src/lib/ecocredit/api.ts

```diff
@@ -163,6 +182,32 @@ export const getBatchWithSupplyForDenom = async (
   }
 };
 
+export const getReadableMessages = (txResponse: TxResponse): string => {
+  return uniq(
+    txResponse?.logs[0]?.events
+      .filter(
+        event =>
+          event.type === 'message' ||
+          Object.values(ECOCREDIT_EVENT_TYPES).some(eventType =>
+            eventType.message.includes(event.type),
+          ),
+      )
+      .map(event => {
+        const eventType =
+          event.attributes.find(attr => attr.key === 'action')?.value ||
+          event.type;
+
+        return getReadableName(eventType);
+      }),
```

Comment:
not sure to get why we're mixing up messages and events here 
why not just look into the tx.body.messages?

<img width="442" alt="image" src="https://user-images.githubusercontent.com/3149358/170219878-49bf6f72-3cbf-43a1-9df6-941e78de926f.png">

Reply:
>why not just look into the tx.body.messages?

This does look like a better place to look, but `txResponses` has the same data. In this use case, I was already using TxResponses for the other table data (not available in `txs`), so it is easier to just continue using that.

I am getting rid of the events. Just messages now.

Reply:
sounds good